lib/FusionInventory/Agent/Task/Inventory/Win32/Softwares.pm

```diff
@@ -316,4 +339,55 @@ sub _processMSIE {
 
 }
 
+# List of SQL Instances
+sub _getSqlInstancesList {
+    my (%params) = @_;
+
+    my $softwareName = $params{softwarename};
+    my $softwareVersion = $params{softwareversion};
+
+    # Registry access for SQL Instances
+    my $sqlinstancesList = getRegistryKey(
+        path    => "HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/Microsoft SQL Server/Instance Names/SQL",
+        %params
+    ) or return;
+
+    # List of SQL Instances
+    foreach my $sqlinstance (keys %$sqlinstancesList) {
+        my $sqlinstanceName = $sqlinstance;
+        my $sqlinstanceValue = $sqlinstancesList->{$sqlinstance};
+        # Get version and edition for each instance
+        my ($sqlinstanceEditionValue,$sqlinstanceVersionValue) = _getSqlInstancesVersions(
+            %params,
+            SOFTVERSION => $softwareVersion,
+            NAME        => $sqlinstanceName,
+            VALUE       => $sqlinstanceValue
+        );
+        next unless $sqlinstanceEditionValue && $sqlinstanceVersionValue;
+        return $sqlinstanceEditionValue
+            if ($softwareVersion eq $sqlinstanceVersionValue); 
+    }
+}
```

Comment:
there is no explicit return value if nothing was found in previous loop.

Reply:
Not really sure, but variable that calls _getSqlEdition function is not defined if nothing is return. Do I returned something if nothing is found ? defined test is not sufficient ?

Reply:
Hum... it seems you're passing $softwareVersion to _getSqlInstancesVersions() and still check the equality there at l. 390. Then I think you don't need to check another time the same equality as $sqlinstanceVersionValue as l. 368 seems for me to be the same than at l. 390 and as $softwareVersion seems not changed between the 2 lines.
If I'm true, you only need to return one value... In that case, you can add `my $sqlinstanceEditionValue;` just before foreach loop begins, then change l. 360 with `$sqlinstanceEditionValue = _getSqlInstancesVersions(`... adapt _getSqlInstancesVersions() API... replace l. 367-368 by `last if defined($sqlinstanceEditionValue);` and insert `return $sqlinstanceEditionValue;` before l. 370.